src/script/components/list/ParticipantItem.tsx

```diff
@@ -0,0 +1,189 @@
+        <div className="participant-item__content__name-wrapper">
+          {isUser && selfInTeam && (
+            <AvailabilityState
+              className="participant-item__content__availability participant-item__content__name"
+              data-uie-name="status-name"
+              availability={(participant as User).availability()}
+              label={ko.unwrap(participant.name)}
+            />
+          )}
+
+          {(isService || !selfInTeam) && (
+            <div className="participant-item__content__name" data-uie-name="status-name">
+              {participant.name}
+            </div>
+          )}
+          {isSelf && <div className="participant-item__content__self-indicator">{selfString}</div>}
+        </div>
+        <div className="participant-item__content__info">
+          {contentInfo && (
+            <Fragment>
+              <span
+                className={cx('participant-item__content__username label-username-notext', {
+                  'label-username': hasUsernameInfo,
+                })}
+                data-uie-name="status-username"
+              >
+                {contentInfo}
+              </span>
+              {hasUsernameInfo && badge && (
+                <span className="participant-item__content__badge" data-uie-name="status-partner">
+                  {badge}
+                </span>
+              )}
+            </Fragment>
+          )}
+        </div>
+      </div>
+      {callParticipant && (
+        <Fragment>
+          {callParticipant.sharesCamera() && <NamedIcon name="camera-icon" data-uie-name="status-video" />}
+          {callParticipant.sharesScreen() && <NamedIcon name="screenshare-icon" data-uie-name="status-screenshare" />}
+
+          {!callParticipant.isMuted() && (
+            <ParticipantMicOnIcon
+              isActive={callParticipant.isActivelySpeaking()}
+              data-uie-name={callParticipant.isActivelySpeaking() ? 'status-active-speaking' : 'status-audio-on'}
+            />
+          )}
+          {callParticipant.isMuted() && <NamedIcon name="mic-off-icon" data-uie-name="status-audio-off" />}
+        </Fragment>
+      )}
+
+      {isUser && !isOthersMode && (participant as User).isGuest() && (
+        <NamedIcon name="guest-icon" data-uie-name="status-guest" />
+      )}
+
+      {external && <NamedIcon name="partner-icon" data-uie-name="status-external" />}
+
+      {isUser && isSelfVerified && (participant as User).is_verified() && (
+        <NamedIcon name="verified-icon" data-uie-name="status-verified" />
+      )}
+
+      {canSelect && (
+        <div
+          className={cx('search-list-item-select icon-check', {selected: isSelected})}
+          data-uie-name="status-selected"
+        ></div>
+      )}
+      <NamedIcon name="disclose-icon" className="disclose-icon" />
+      {/* END isInViewport() */}
+    </div>
+  );
+};
+
+export default ParticipantItem;
+
+registerReactComponent('participant-item', {
```

Comment:
Florian added generic types to `registerReactComponent`, so I guess you can do `registerReactComponent<ParticipantItemProps>` now?!